chore: use prettier instead of clang-format

maribethb opened the pull request:
<!--
  - Thanks for submitting code to Blockly!  Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## The basics

<!-- TODO: Verify the following, checking each box with an 'x' between the brackets: [x] -->

- [x] I branched from develop
- [X] My pull request is against develop
- [X] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)
- [X] I ran `npm run format` and `npm run lint`

## The details
### Resolves

<!-- TODO: What Github issue does this resolve? Please include a link. -->
Fixes #6993 (by using prettier instead of clang-format)

### Proposed Changes

This PR best reviewed commit-by-commit:

- Adds Prettier tool and config; used eslint-prettier-config to remove conflicting eslint rules
- Removes clang-format tool, config, and references
- Reformats all files in project
    - Removed all clang-format off comments, only added two prettier-disable comments. Prettier treated some lines that had to be ignored by clang a lot better and we don't have to ignore them.
- Updates eslint config to also lint ts files outside of `core/`
- Fixes newly revealed lint problems in `blocks/`  

#### Behavior Before Change

#### Behavior After Change

No actual behavior changes.

Notable format changes:
- Space after `function` keyword in anonymous functions. This makes it consistent that there is always a space after the `function` keyword just like every other keyword
- If there are too many parameters to fit on one line, each one gets its own line, with closing bracket also on own line
- Multiple destructured imports are one per line

### Reason for Changes

Clang-format doesn't handle TypeScript well. We were having to disable it entirely in the `blocks/` directory. Indeed, there is even an [issue](https://github.com/google/blockly/issues/6920) where we note something that clang-format mangled. Prettier handles this better and is more readable.

### Test Coverage

Tests pass. Manual inspection of files. I spot-checked several hundred files for formatting problems.

Pushed a commit with bad format to make sure the github action worked and it [did](https://github.com/google/blockly/actions/runs/4802367751/jobs/8545743702?pr=7014).

### Documentation

<!-- TODO: Does any documentation need to be created or updated because of this PR?
  -        If so please explain.
  -->

### Additional Information

<!-- Anything else we should know? -->

maribethb committed:
chore: add and configure prettier

maribethb committed:
chore: remove clang-format

maribethb committed:
chore: remove clang-format config

maribethb committed:
chore: lint additional ts files

maribethb committed:
chore: fix lint errors in blocks

rachel-fenichel commented on blocks/lists.ts:
```diff
@@ -361,12 +360,10 @@ const LISTS_INDEXOF = {
-    // Assign 'this' to a variable for use in the tooltip closure below.
-    const thisBlock = this;
-    this.setTooltip(function () {
+    this.setTooltip(() => {
```
Did prettier rewrite this into an arrow function, or did you do that?

maribethb replied:
I did this, to resolve lint errors related to redefining `this`

rachel-fenichel replied:
Oh good. I was not stoked about automated rewriting of functions into arrow functions.

rachel-fenichel commented on .github/workflows/assign_reviewers.yml:
```diff
@@ -16,26 +16,26 @@ jobs:
-            if (context.payload.pull_request === undefined) {
-            throw new Error("Can't get pull_request payload. " +
-                            'Check a request reviewer event was triggered.');
+      - name: Assign requested reviewer
```
Prettier changed indentation in a bunch of yaml files. Please confirm that it's still file/the indentation is irrelevant.

rachel-fenichel reviewed (commented):
Two questions, but overall lgtm and I'm happy to get rid of our use of that random `clang-format` github action.

rachel-fenichel commented:
LGTM once you fix conflicts

maribethb committed:
chore: add prettier-ignore where needed

maribethb committed:
chore: ignore js blocks when formatting

maribethb committed:
chore: fix playground html syntax

maribethb committed:
chore: fix yaml spacing from merge

maribethb committed:
chore: convert text blocks to use arrow functions

maribethb committed:
chore: format everything with prettier

maribethb committed:
chore: fix lint unused imports in blocks

maribethb commented:
I've fixed conflicts, re-run prettier after rebasing, and removed the requirement of `clang-format` from the develop branch. After pressing merge I'll require the new action instead.
